profiler/src/profile_convnd_bwd_data.cpp

```diff
@@ -0,0 +1,218 @@
+    {
+        params.conv_filter_dilations[i] = std::stoi(argv[arg_idx++]);
+    }
+    params.input_left_pads.resize(num_dim_spatial);
+    for(int i = 0; i < num_dim_spatial; ++i)
+    {
+        params.input_left_pads[i] = std::stoi(argv[arg_idx++]);
+    }
+    params.input_right_pads.resize(num_dim_spatial);
+    for(int i = 0; i < num_dim_spatial; ++i)
+    {
+        params.input_right_pads[i] = std::stoi(argv[arg_idx++]);
+    }
+
+    return params;
+}
+
+int profile_convnd_bwd_data(int argc, char* argv[], int num_dim_spatial)
+{
+    const int preParams = 10;
+    int conv_args       = 3 + num_dim_spatial * 6;
+    int cmdline_nargs   = conv_args + preParams;
+    if(cmdline_nargs != argc)
+    {
+        printf("arg1: tensor operation (conv[1|2|3]d_bwd_data: BackwardConvolution)\n");
+        printf("arg2: data type (0: fp32; 1: fp16)\n");
+        printf("arg3: input tensor layout (0: NCHW; 1: NHWC)\n");
+        printf("arg4: weight tensor layout (0: KCYX; 1: KYXC)\n");
+        printf("arg5: output tensor layout (0: NKHW; 1: NHWK)\n");
+        printf("arg6: verification (0: no; 1: yes)\n");
+        printf("arg7: initialization (0: no init; 1: integer value; 2: decimal value)\n");
+        printf("arg8: print tensor value (0: no; 1: yes)\n");
+        printf("arg9: run kernel # of times (>1)\n");
+        printf("arg10 to 24: N, K, C, Y, X, Hi, Wi, Sy, Sx, Dy, Dx, LeftPy, LeftPx, RightPy, "
+               "RightPx\n");
+        exit(1);
+    }
+
+    const int data_type        = static_cast<ConvDataType>(std::stoi(argv[2]));
+    const int in_layout        = static_cast<ConvInputLayout>(std::stoi(argv[3]));
+    const int wei_layout       = static_cast<ConvWeightLayout>(std::stoi(argv[4]));
+    const int out_layout       = static_cast<ConvOutputLayout>(std::stoi(argv[5]));
+    const bool do_verification = std::stoi(argv[6]);
+    const int init_method      = std::stoi(argv[7]);
+    const bool do_log          = std::stoi(argv[8]);
+    const int nrepeat          = std::stoi(argv[9]);
+
+    ck::conv_util::ConvParams params = ParseConvParams(num_dim_spatial, argv, preParams);
+
+    auto Run = [&](auto input_type, auto wei_type, auto out_type) {
+        using InDataType  = decltype(input_type);
+        using WeiDataType = decltype(wei_type);
+        using OutDataType = decltype(out_type);
+
+        switch(num_dim_spatial)
+        {
+        case 1:
+            ck::profiler::profile_convnd_bwd_data_impl<1,
+                                                       InDataType,
+                                                       WeiDataType,
+                                                       OutDataType,
+                                                       ck::tensor_layout::convolution::NHWC,
+                                                       ck::tensor_layout::convolution::KYXC,
+                                                       ck::tensor_layout::convolution::NHWK>(
```

Comment:
NWC, KXC, NWK.

The reason why this compiles is because in `profile_convnd_bwd_data_impl` the layout info is not used at all. See the corresponding comment.